examples/dpo_demo_gemma3.ipynb

```diff
@@ -0,0 +1,849 @@
+    "              # passed to model forward pass\n",
+    "              \"prompts\": TEMPLATE.format(\n",
+    "                  system_prompt=SYSTEM_PROMPT,\n",
+    "                  question=x[\"question\"].decode(\"utf-8\"),\n",
+    "              ),\n",
+    "              # passed to reward functions\n",
+    "              \"question\": x[\"question\"].decode(\"utf-8\"),\n",
+    "              # passed to reward functions\n",
+    "              \"answer\": extract_hash_answer(x[\"answer\"].decode(\"utf-8\")),\n",
+    "          }\n",
+    "      )\n",
+    "  )\n",
+    "  return dataset\n",
+    "\n",
+    "test_dataset = get_dataset(TEST_DATA_DIR, \"test\").batch(BATCH_SIZE)[\n",
+    "    :NUM_TEST_BATCHES\n",
+    "]\n",
+    "\n",
+    "len(test_dataset)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(corr, total, accuracy, partial_accuracy, format_accuracy), responses = evaluate(\n",
+    "    test_dataset,\n",
+    "    sampler,\n",
+    "    **GENERATION_CONFIGS[\"standard\"],\n",
+    "    make_lst=True\n",
+    ")\n",
+    "print(\n",
+    "    f\"{corr=}, {total=}, {accuracy=}%, {partial_accuracy=}%,\"\n",
+    "    f\" {format_accuracy=}%\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "LCAiINxUwYqx"
+   },
+   "source": [
+    "# Load DPO dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dpo_dataset = load_dataset(\"argilla/distilabel-intel-orca-dpo-pairs\", split=\"train\")\n",
+    "gsm8k_train_dpo_dataset = dpo_dataset.filter(lambda x: x['in_gsm8k_train'])\n",
+    "\n",
+    "# Get the number of samples in the filtered dataset\n",
+    "num_gsm8k_train_samples = len(gsm8k_train_dpo_dataset)\n",
+    "print(f\"Number of samples with in_gsm8k_train=True: {num_gsm8k_train_samples}\")\n",
+    "\n",
+    "# Calculate how many more samples are needed\n",
+    "total_samples_needed = NUM_BATCHES\n",
+    "samples_to_add = total_samples_needed - num_gsm8k_train_samples\n",
+    "print(f\"Number of additional random samples needed: {samples_to_add}\")\n",
+    "\n",
+    "if samples_to_add > 0:\n",
+    "    # Randomly select additional samples from the original dataset\n",
+    "    # Ensure we don't sample more than the total available in the original dataset\n",
+    "    random_samples = dpo_dataset.shuffle(seed=42).select(range(min(samples_to_add, len(dpo_dataset))))\n",
+    "    print(f\"Number of random samples selected: {len(random_samples)}\")\n",
+    "\n",
+    "    # Combine the filtered dataset and the random samples\n",
+    "    combined_dpo_dataset = concatenate_datasets([gsm8k_train_dpo_dataset, random_samples])\n",
+    "else:\n",
+    "    combined_dpo_dataset = gsm8k_train_dpo_dataset\n",
+    "\n",
+    "print(f\"Total samples in the combined dataset: {len(combined_dpo_dataset)}\")\n",
```

Comment:
Okay, so here, we keep all samples present in GSM8K and add a few non-GSM8K samples randomly, right?

Good to add that we are trying to do the above here, and also explain the purpose of it. We can add it to a markdown cell